rs/nns/integration_tests/src/test_proposal_with_golden_nns_state.rs

```diff
@@ -0,0 +1,107 @@
+use std::env;
+
+use ic_base_types::PrincipalId;
+use ic_ledger_core::tokens::Tokens;
+use ic_nns_common::pb::v1::ProposalId;
+use ic_nns_test_utils::{
+    nns_canister_upgrade::NnsCanisterUpgrade,
+    state_test_helpers::{
+        adopt_proposal, nns_create_neuron_with_stake, nns_create_super_powerful_neuron,
+        nns_get_network_economics_parameters, nns_propose_upgrade_nns_canister,
+        nns_wait_for_proposal_execution, wait_for_canister_upgrade_to_succeed,
+    },
+};
+use ic_nns_test_utils_golden_nns_state::new_state_machine_with_golden_nns_state_or_panic;
+
+#[test]
+fn test_proposal_with_golden_nns_state() {
+    // Step 1: Prepare the world
+
+    // Make sure we can get a test version of NNS Governance. This enables adopting open proposals.
+    let mut nns_canister_upgrade = NnsCanisterUpgrade::new("governance-test");
+
+    // Step 1.1: Load golden nns state into a StateMachine.
+    let state_machine = new_state_machine_with_golden_nns_state_or_panic();
+
+    // Save initial state for the ultimate assertion of this test.
+    let network_economics = nns_get_network_economics_parameters(&state_machine);
+
+    // Phase I. Upgrade NNS Governance to a test-only version (using a super powerful neuron) that
+    // allows adopting any proposal by calling adopt_proposal.
+    {
+        let neuron_controller = PrincipalId::new_self_authenticating(&[1, 2, 3, 4]);
+        let proposer_neuron_id =
+            nns_create_super_powerful_neuron(&state_machine, neuron_controller);
+
+        println!("Proposing to upgrade NNS Governance ...");
+
+        let _proposal_id = nns_propose_upgrade_nns_canister(
+            &state_machine,
+            neuron_controller,
+            proposer_neuron_id,
+            nns_canister_upgrade.canister_id,
+            nns_canister_upgrade.wasm_content.clone(),
+            nns_canister_upgrade.module_arg.clone(),
+            true,
+        );
+
+        // Step 3: Verify result(s): In a short while, the canister should be running the new code.
```

Comment:
I would say that this is part of setting up the world (step 1). At this point, we are not yet verifying the code under test. (We are merely installing the code under test.)